nasa/harmony-browse-image-generator
# DAS-2259 fix opera rtc browse images.

## Description

A well placed, careful, cast to uint8, prevents trying to create PIL images from array with float32 values.

Also updates numpy to v2. because it has the np.min_scalar_type used in debugging.


## Jira Issue ID

[DAS-2259](https://bugs.earthdata.nasa.gov/browse/DAS-2259)

## Local Test Steps

- Check out this branch, build and run the tests.

``` shell
❯ ./bin/build-image && ./bin/build-test && ./bin/run-test
```

- Verify all tests pass.  Fire up Harmony-In-A-Box and run the regression test notebook against localhost. 

- Examine the notebook explain-opera.pdf attached to the DAS-2259 ticket for explanation of the change.

- Lastly generate an image for OPERA RTC:

	1. Download the input tif on the DAS-2259 ticket
   OPERA_L2_RTC-S1_T039-082699-IW3_20240909T092236Z_20240909T134139Z_S1A_30_v1.0_rgb.tif
 
	2. Run the script below (after installing hybig as editable in your environment)
    ``` python
     from hybig import create_browse
     fn = './OPERA_L2_RTC-S1_T039-082699-IW3_20240909T092236Z_20240909T134139Z_S1A_30_v1.0_rgb.tif'
     create_browse(fn, {})
    ```
	3. bask in the glow of output PNGs.




## PR Acceptance Checklist
